src/api/java/appeng/api/implementations/ICraftingPatternItem.java

```diff
@@ -23,24 +23,75 @@
 
 package appeng.api.implementations;
 
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
-import net.minecraft.world.World;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
 
-import appeng.api.networking.crafting.ICraftingPatternDetails;
+import appeng.api.storage.data.IAEItemStack;
 
 /**
- * Implemented on {@link Item}
+ * Implemented on a {@link Item}.
+ * 
+ * All methods should accept an {@link ItemStack} to provide the actual data.
```

Comment:
```suggestion
```

Reply:
Potentially, a mod could add an encoded pattern item that represent a single recipe.